crates/biome_grit_patterns/src/grit_target_language.rs

```diff
@@ -1,25 +1,94 @@
+/// creating language-specific implementations.
+macro_rules! generate_target_language {
+    ($($language:ident),+) => {
+        #[derive(Clone, Debug)]
+        pub enum GritTargetLanguage {
+            $($language($language)),+
+        }
 
-impl Language for GritTargetLanguage {
-    type Node<'a> = GritNode;
+        $(impl From<$language> for GritTargetLanguage {
+            fn from(value: $language) -> Self {
+                Self::$language(value)
+            }
+        })+
 
-    fn language_name(&self) -> &'static str {
-        todo!()
-    }
+        impl GritTargetLanguage {
+            fn metavariable_kind(&self) -> GritTargetSyntaxKind {
+                match self {
+                    $(Self::$language(_) => $language::metavariable_kind().into()),+
+                }
+            }
 
-    fn snippet_context_strings(&self) -> &[(&'static str, &'static str)] {
-        todo!()
-    }
+            fn is_alternative_metavariable_kind(&self, kind: GritTargetSyntaxKind) -> bool {
+                match self {
+                    $(Self::$language(_) => $language::is_alternative_metavariable_kind(kind)),+
+                }
+            }
+        }
 
-    fn is_comment(&self, _node: &Self::Node<'_>) -> bool {
-        todo!()
-    }
+        impl Language for GritTargetLanguage {
+            type Node<'a> = GritTargetNode;
+
+            fn language_name(&self) -> &'static str {
+                match self {
+                    $(Self::$language(language) => language.language_name()),+
+                }
+            }
+
+            fn snippet_context_strings(&self) -> &[(&'static str, &'static str)] {
+                match self {
+                    $(Self::$language(language) => language.snippet_context_strings()),+
+                }
+            }
+
+            fn is_comment(&self, node: &GritTargetNode) -> bool {
+                match self {
+                    $(Self::$language(language) => language.is_comment(node)),+
+                }
+            }
+
+            fn is_metavariable(&self, node: &GritTargetNode) -> bool {
+                node.kind() == self.metavariable_kind()
+                    || (self.is_alternative_metavariable_kind(node.kind())
+                        && self.exact_replaced_variable_regex().is_match(&node.text().to_string()))
+            }
+        }
+    };
+}
+
+generate_target_language! {
+    JsTargetLanguage
+}
+
+/// Trait to be implemented by the language-specific implementations.
+///
+/// This is used to make language implementations a little easier, by not
+/// forcing them to reimplement methods that are common across implementations.
+trait GritTargetLanguageImpl {
+    type Kind: SyntaxKind;
+
+    fn language_name(&self) -> &'static str;
+
+    fn snippet_context_strings(&self) -> &[(&'static str, &'static str)];
+
+    fn is_comment(&self, node: &GritTargetNode) -> bool;
+
+    fn metavariable_kind() -> Self::Kind;
```

Comment:
Can we document the functions? For example, I checked `snippet_context_strings` and really can't understand what it is.